### PR TITLE
Improve Logical Flow of Documentation

### PR DESCRIPTION
**Pull request by CSSFrancis**
### Description of the change

I feel like I always get lost in the Rosettasciio documentation.  Most of the time I just want to go to this page https://hyperspy.org/rosettasciio/user_guide/supported_formats/index.html but it's always like 2-3 levels down. I also
almost end up going to the File Specifications page first.   

Rosettasciio has a lot of documentation but it's not apparent from the home page :) 

### Progress of the PR
- [x] Grid to Front Page
- [ ] Make User Guide Look Cleaner
- [ ] ready for review.

### CODE REVIEW

**Comment by jlaehne** [comment]
Very good initiative, I have the same experience. But lets have a grid directly on the start page (similar to https://hyperspy.org/exspy/ ), so that one can get to the supported formats with one click!

**Comment by jlaehne** [comment]
I think you need to add the extension `"sphinx_design"` to make the grid work.

File: doc/index.rst [hunk 1]
```diff
@@ -66,6 +66,7 @@ Table of contents
 
    user_guide/index
    file_specification/index
+   user_guide/supported_formats
```
Comment (jlaehne) [inline doc/index.rst:69]: Does not work currently, and is already a subcategory of user_guide. Instead add a grid to the main page with a prominent link to the table of supported formats (Maybe even before installation and any other category).

Comment (jlaehne) [reply doc/index.rst:69]: How about a grid with six items:

* Supported Formats
* Installation
* Interoperability

and in a second row:

* File specifications
* API
* Contributing

That could replace the whole TOC and come before the splitting note and the citing section!

**Comment by CSSFrancis** [comment]
@jlaehne What do you think the User Guide should show?  It might be nice to just jump straight to installation and have people use the side bar for something else?

**Comment by jlaehne** [comment]
> @jlaehne What do you think the User Guide should show? It might be nice to just jump straight to installation and have people use the side bar for something else?

As all sections are now accessible from the front page as well that would indeed make sense.

Also, I think the table of contents on the front page could be dropped now.

Finally, I would put API before file specifications both in the header and the grid.

File: doc/index.rst [hunk 2]
```diff
@@ -19,6 +19,66 @@ project for multi-dimensional data analysis. As HyperSpy is rooted in the electr
...
+    Documentation describing how to use rosettasciio with other libraries than HyperSpy.
+
+  .. grid-item-card::
+    :link: file_specification/index
+    :link-type: doc
+
+    :octicon:`checklist;2em;sd-text-info` File Specifications
+    ^^^
+
+    Links to File Specifications for the different file formats supported by rosettasciio.
```
Comment (jlaehne) [inline doc/index.rst:61]:
```suggestion
    Links to or documentation of file specifications for some of the formats supported by RosettaSciIO.
```

File: doc/index.rst [hunk 3]
```diff
@@ -19,6 +19,66 @@ project for multi-dimensional data analysis. As HyperSpy is rooted in the electr
...
+    Documentation of the Application Programming Interface (API),
+
+  .. grid-item-card::
+    :link: contributing
+    :link-type: doc
+
+    :octicon:`code-square;2em;sd-text-info` Contributing
+    ^^^
+
+    Information on how to contribute to the development of rosettasciio.
```
Comment (jlaehne) [inline doc/index.rst:79]:
```suggestion
    Information on how to implement new file plugins or help improve existing ones.
```

File: doc/index.rst [hunk 4]
```diff
@@ -19,6 +19,66 @@ project for multi-dimensional data analysis. As HyperSpy is rooted in the electr
...
+
+
 RosettaSciIO provides the dataset, its axes and related metadata contained in a
 file in a python dictionary that can be easily handled by other libraries.
```
Comment (jlaehne) [inline doc/index.rst:83]:
```suggestion
file in a :ref:`python dictionary <interfacing-api>` that can be easily handled by other libraries.
```

**Review by ericpre — COMMENTED** [review]
Thank you @CSSFrancis, this has been bugging me for a while too and these changes are useful!
Most of the time, what we need to access is the "Supported formats" section and it would much better to have the link in the top bar too. I don't know if this is possible without taking that section out of the "User guide" section - in which case, the user guide could be simplify to installation, a couple of example on how to and link to the Supported formats" section.

File: doc/index.rst [hunk 5]
```diff
@@ -19,9 +19,74 @@ project for multi-dimensional data analysis. As HyperSpy is rooted in the electr
...
+
+  .. grid-item-card::
+    :link: user_guide/interoperability
+    :link-type: doc
+
+    :octicon:`arrow-switch;2em;sd-text-info` Interoperability
+    ^^^
+
+   Information on scripts and packages that allow opening data saved using
+   RosettaSciIO with third-party programs.
```
Comment (jlaehne) [inline doc/index.rst:54]:
```suggestion
    Information on scripts and packages that allow opening data saved using
    RosettaSciIO with third-party programs.
```

File: doc/index.rst [hunk 6]
```diff
@@ -19,9 +19,74 @@ project for multi-dimensional data analysis. As HyperSpy is rooted in the electr
...
+
+  .. grid-item-card::
+    :link: file_specification/index
+    :link-type: doc
+
+    :octicon:`checklist;2em;sd-text-info` File Specifications
+    ^^^
+
+     Links to or documentation of file specifications for
+     some of the formats supported by RosettaSciIO.
```
Comment (jlaehne) [inline doc/index.rst:64]:
```suggestion
    Links to or documentation of file specifications for
    some of the formats supported by RosettaSciIO.
```

**Review by jlaehne — COMMENTED** [review]
There are some formatting issues in the preview that should be related to indentation errors.

**Comment by jlaehne** [comment]
Do you want to fix the formatting issues and conflicts. Would be nice to have this one in soon :-)

**Comment by CSSFrancis** [comment]
@jlaehne it should be good now!

**Review by ericpre — COMMENTED** [review]
The documentation warning needs to be fixed to make the documentation build pass. 

Should the further improvement with the user guide done in a follow up PR?

**Review by ericpre — APPROVED** [review]

**Comment by ericpre** [comment]
Failure are unrelated

**Pull request merged**